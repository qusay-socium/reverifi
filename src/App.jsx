import AdminPanelLayout from 'components/shared/AdminPanelLayout';
import MainLayout from 'components/shared/MainLayout';
import { UserProvider } from 'contexts/UserContext';
import React from 'react';
import { Route, Routes } from 'react-router-dom';
import routes from 'routes';
import GlobalStyles from 'styles/global-styles';

/**
 * App entry point component.
 *
 * @return {JSX.Element}
 */
function App() {
  return (
    <UserProvider>
      <GlobalStyles />
      <React.Suspense fallback={<span>Loading...</span>}>
        <Routes>
          {routes.map(
            ({
              label,
              exact,
              path,
              component: Component,
              adminLayout,
              disableNavbar,
              disableFooter,
            }) => (
              <Route
                key={label}
                path={path}
                exact={exact}
                element={
                  !adminLayout ? (
                    <MainLayout
                      disableNavbar={disableNavbar}
                      disableFooter={disableFooter}
                    >
                      <Component />
                    </MainLayout>
                  ) : (
                    <AdminPanelLayout>
                      <Component />
                    </AdminPanelLayout>
                  )
                }
              />
            )
          )}
        </Routes>
      </React.Suspense>
    </UserProvider>
  );
}

export default App;
