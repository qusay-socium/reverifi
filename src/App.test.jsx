import React from 'react';
import { shallow } from 'enzyme';
import App from 'App';

describe('Events Section Testing', () => {
  it('Renders Correctly', () => {
    const Wrapper = shallow(<App />);
    expect(Wrapper).toMatchSnapshot();
  });
});
