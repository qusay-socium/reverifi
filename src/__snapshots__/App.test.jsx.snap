// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Section Testing Renders Correctly 1`] = `ShallowWrapper {}`;
